src/time_integration/methods_IMEXJinXin.jl

```diff
@@ -0,0 +1,728 @@
+            terminate!(integrator)
+        end
+    end
+
+    # Empty the tstops array.
+    # This cannot be done in terminate!(integrator::SimpleIntegratorSSP) because DiffEqCallbacks.PeriodicCallbackAffect would return at error.
+    extract_all!(integrator.opts.tstops)
+
+    for stage_callback in alg.stage_callbacks
+        finalize_callback(stage_callback, integrator.p)
+    end
+
+    return TimeIntegratorSolution((first(prob.tspan), integrator.t),
+                                  (prob.u0, integrator.u), prob)
+end
+
+function divide_relaxed_var!(u, dt, semi, solver, cache, aii, equations, mesh)
+    u_wrap = Trixi.wrap_array(u, semi)
+    cycle_divide!(u_wrap, dt, semi, solver, cache, aii, equations, mesh)
+    return nothing
+end
+
+function cycle_divide!(u, dt, semi, solver, cache, aii, equations, mesh::TreeMesh2D)
+    @unpack inverse_jacobian = cache.elements
+    nvars_base = nvariables(equations.equations_base)
+    relaxation_rate = equations.eps_relaxation
+    for element in eachelement(solver, cache)
+        factor = inverse_jacobian[element]
+        #factor = 1.0
+        for j in eachnode(solver), i in eachnode(solver)
+            for var in (nvars_base + 1):(nvars_base * 3)
+                u[var, i, j, element] = u[var, i, j, element] /
+                                        (1.0 + factor * dt / relaxation_rate * aii)
+            end
+        end
+    end
+
+    return nothing
+end
+
+function cycle_divide!(u, dt, semi, solver, cache, aii, equations, mesh::TreeMesh1D)
+    @unpack inverse_jacobian = cache.elements
+    nvars_base = nvariables(equations.equations_base)
+    relaxation_rate = equations.eps_relaxation
+    for element in eachelement(solver, cache)
+        factor = inverse_jacobian[element]
+        factor = 1.0
+        for i in eachnode(solver)
+            for var in (nvars_base + 1):(nvars_base * 2)
+                u[var, i, element] = u[var, i, element] /
+                                     (1.0 + factor * dt / relaxation_rate * aii)
+            end
+        end
+    end
+
+    return nothing
+end
+
+function wrap_and_perform_projection!(u, dt, mesh, equations, solver, cache)
+    u_wrap = wrap_array(u, mesh, equations, solver, cache)
+    perform_projection_sourceterm!(u_wrap, dt, mesh, equations, solver, cache)
+
+    return nothing
+end
+
+function set_cons_var_to_zero!(u, semi, solver, cache, equations, mesh::TreeMesh2D)
+    u_wrap = Trixi.wrap_array(u, semi)
+    nvars_base = nvariables(equations.equations_base)
+    @unpack inverse_jacobian = cache.elements
+    for element in eachelement(solver, cache)
+        factor = inverse_jacobian[element]
+        #   factor = 1.0
+        for j in eachnode(solver), i in eachnode(solver)
+            for var in 1:nvars_base
+                u_wrap[var, i, j, element] = 0.0
+            end
+            for var in (nvars_base + 1):(nvars_base * 3)
+                u_wrap[var, i, j, element] *= factor
+            end
```

Comment:
This function is called `set_to_zero!` but also multiplies the other variables by some `factor`. Is this desired? If so, can we rename the function accordingly?